config/routes.rb

```diff
@@ -149,6 +149,12 @@
     end
   end
 
+  namespace :api do
+    get 'generic_works/:id' => 'generic_works#show'
+    get 'generic_works/:id/add_parent/:parent_id' => 'generic_works#add_parent'
+    get 'generic_works/:id/remove_parent/:parent_id' => 'generic_works#remove_parent'
```

Comment:
You should use a `delete` action, not a `get` for request that removes a resource.